crates/text_edit/src/lib.rs

```diff
@@ -16,6 +18,22 @@ pub struct Indel {
     pub delete: TextRange,
 }
 
+impl PartialOrd for Indel {
+    fn partial_cmp(&self, other: &Indel) -> Option<Ordering> {
+        if self == other {
+            return Some(Ordering::Equal);
+        }
+        let cmp_to_other = self.delete.end() <= other.delete.start();
+        let cmp_to_self = other.delete.end() <= self.delete.start();
+        match (cmp_to_other, cmp_to_self) {
+            (true, true) => None,
```

Comment:
This would mean that insertions with the same offset but different insertions are to be seen as not equal. This is why I introduce `eq_insert_at_offset` to treat that case when it is necessary